yamllint/config.py

```diff
@@ -179,16 +179,16 @@ def validate_rule_conf(rule, conf):
                 continue
             if optkey not in options:
                 raise YamlLintConfigError(
-                    'invalid config: unknown option "%s" for rule "%s"' %
-                    (optkey, rule.ID))
+                    f'invalid config: unknown option "{optkey}" '
+                    f'for rule "{rule.ID}"')
             # Example: CONF = {option: (bool, 'mixed')}
             #          → {option: true}         → {option: mixed}
             if isinstance(options[optkey], tuple):
                 if (conf[optkey] not in options[optkey] and
                         type(conf[optkey]) not in options[optkey]):
                     raise YamlLintConfigError(
-                        'invalid config: option "%s" of "%s" should be in %s'
-                        % (optkey, rule.ID, options[optkey]))
+                        f'invalid config: option "{optkey}" of "{rule.ID} '
+                        f'should be in {options[optkey]}')
```

Comment:
There's a missing `"` after `{rule.ID}`, making the tests fail:

```
FAIL: test_yes_no_for_booleans (tests.test_config.SimpleConfigTestCase)
----------------------------------------------------------------------
yamllint.config.YamlLintConfigError: invalid config: option "indent-sequences" of "indentation should be in (<class 'bool'>, 'whatever', 'consistent')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/work/yamllint/yamllint/tests/test_config.py", line 109, in test_yes_no_for_booleans
    c = config.YamlLintConfig('rules:\n'
AssertionError: "invalid config: option "indent-sequences" of "indentation" should be in " does not match "invalid config: option "indent-sequences" of "indentation should be in (<class 'bool'>, 'whatever', 'consistent')"
```

Reply:
You're right about UTF-8, I will revert that:
https://docs.python.org/3/library/functions.html#open

Reply:
Let me have a look at the rest of the formatting.